_source/_posts/2021-02-01-building-a-github-secrets-scanner.md

```diff
@@ -0,0 +1,219 @@
+---
+layout: blog_post
+title: Building a GitHub Secrets Scanner
+author: vickie-li
+by: contractor
+communities: [python]
+description: "Use Python to build a basic GitHub secrets scanner."
+tags: [python]
+tweets:
+- "Learn the basics of how to scan GitHub for sensitive information."
+- "Would you know if someone checked secrets into GitHub? Learn how to build a tool to check."
+image: blog/building-a-github-secrets-scanner/card.png
+type: awareness
+---
+GitHub reconnaissance is a tactic that attackers use to gather information about their targets. Attackers analyze organizations' GitHub repositories and check for sensitive data that has been accidentally committed or information that could lead to the discovery of a vulnerability.
+For this tutorial, let's build a scanner that automates the GitHub recon process! We will be scanning your GitHub repositories using the method mentioned in the "Tightening Up Your GitHub Security" post.
+We will be working with Python, the GitHub REST API, and the GitPython Library. You will need to obtain a [GitHub personal access token](https://github.com/settings/tokens) and [install GitPython](https://gitpython.readthedocs.io/en/stable/intro.html#installing-gitpython). 
```

Comment:
```suggestion

You will be working with Python, the GitHub REST API, and the GitPython Library. You will need to obtain a [GitHub personal access token](https://github.com/settings/tokens) and [install GitPython](https://gitpython.readthedocs.io/en/stable/intro.html#installing-gitpython). 
```

I could also see these items listed in a prerequisites bullet-list like we do in other posts.